tests/installation/system_role.pm

```diff
@@ -68,7 +68,14 @@ sub assert_system_role {
 }
 
 sub run {
-    if (is_sle('<15') && !is_x86_64) {
+    if (is_sle('=12-sp5') && is_sles4sap) {
+        # Because is_sles4sap depends on the openQA settings and does not
+        # interogate the SUT to ask for the actual status, we need an extra
+        # needle check to make sure we are in the correct place and the System Role
+        # dialog did not come up.
+        assert_screen('partitioning-edit-proposal-button');
+    }
```

Comment:
Similar to the first `elsif`, I think it'd be a good idea to add a `record_info` here indicating that test is skipping the screen/check because there is no system role selection in SLES for SAP 12-SP5

Reply:
I will add the record_info

Reply:
I will not remove the needle check because the comments reminded me why I added it. It exists because when `system_role.pm` is run in create_hdd_*SLES*_gnome, the job we need for migrations,  `is_sles4sap` returns true. So we need to add a check_screen to make sure the system role *SCREEN* has not come up.

Here is an example where I run `is_sles4sap` in the sles create_hdd job and it returns true. http://openqaworker15.qa.suse.cz/tests/295826#step/system_role/1